src/hooks/useGlobalAppState.ts

```diff
@@ -142,6 +77,29 @@ const onStateChange = (
  */
 export const useGlobalAppState = (): void => {
   const client = useClient()
+  const initFlag = useRef(false)
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+  const routeIndex = useNavigationState(state => state?.index)
+
+  // On app start
+  useEffect(() => {
+    log.info('useGlobalAppState: app start')
+
+    if (getIsSecurityFlowPassed() && !initFlag.current) {
+      log.info('useGlobalAppState: app start, security flow passed')
+
+      if (isSharingMode()) {
+        log.info('useGlobalAppState: app start, sharing mode')
+        navigate(routes.sharing)
+      } else {
+        log.info('useGlobalAppState: app start, not sharing mode')
+      }
+
+      initFlag.current = true
+    } else {
+      log.info('useGlobalAppState: app start, security flow not passed')
+    }
+  }, [routeIndex])
```

Comment:
Does this mean that this effect will be called on every navigation (i.e. when home is opening a cozy-app)?

Reply:
yes, it's not great, but I need to listen to something. Maybe I could replace it with an event listener